File: backends/js/src/regenerate-api.ts
#!/usr/bin/env node

import { compile, type JavaScriptIDLCompilerOptions } from "./index.js";
import * as path from "node:path";


const dir = import.meta.dirname;
console.log(dir);

const compilerOptions: JavaScriptIDLCompilerOptions = {
	languageOptions: {
		outputDir: path.join(dir, "../src"),
		packageName: "@argon-lang/noble-idl-compiler-js",
		packageOptions: new Map([
			[ "@argon-lang/noble-idl-runtime", {
				packageMapping: new Map([
					[ "nobleidl.core", "" ],
				]),
			} ],
			[ "@argon-lang/noble-idl-api", {
				packageMapping: new Map([
					[ "nobleidl.compiler.api", "api" ],
				]),
			} ],
		]),
	},

	inputFiles: [
		path.join(dir, "../../noble-idl/compiler-api.nidl"),
	],

	libraryFiles: [
		path.join(dir, "../../../runtime/noble-idl/nobleidl-core.nidl"),
	],
};
await compile(compilerOptions);

File: langs/java/backend/src/main/java/dev/argon/nobleidl/compiler/RegenerateApi.java
package dev.argon.nobleidl.compiler;

import dev.argon.esexpr.KeywordMapping;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.Map;

public class RegenerateApi {

	public static void main(String[] args) throws Throwable {
		String coreLib = Files.readString(Path.of("../../noble-idl/runtime/nobleidl-core.nidl"));
		String compilerApi = Files.readString(Path.of("../../noble-idl/backend/compiler-api.nidl"));

		JavaNobleIDLCompiler.compile(new JavaIDLCompilerOptions(
			new JavaLanguageOptions(
				"src/gen/java",

				new PackageMapping(
					new KeywordMapping<>(Map.of(
						"nobleidl.core", "dev.argon.nobleidl.runtime",
						"nobleidl.compiler.api", "dev.argon.nobleidl.compiler.api"
					))
				)
			),
			List.of(compilerApi),
			List.of(coreLib)
		));
	}

}
